tests/functional/demo/test_demo.py

```diff
@@ -538,3 +538,31 @@ def test_demo_with_nondefault_config_keyword(database, monkeypatch):
     assert (params[0]['value'] - 34.56789) < 1e-5
 
     orion.core.config.user_script_config = 'config'
+
+
+@pytest.mark.usefixtures("clean_db")
+@pytest.mark.usefixtures("null_db_instances")
+def test_demo_precision(database, monkeypatch):
+    """Test a simple usage scenario."""
+    monkeypatch.chdir(os.path.dirname(os.path.abspath(__file__)))
+
+    user_args = [
+        "-x~uniform(-50, 50, precision=5)",
+        "--test-env",
+        "--experiment-id", '{exp.id}',
+        "--experiment-name", '{exp.name}',
+        "--experiment-version", '{exp.version}',
+        "--trial-id", '{trial.id}',
+        "--working-dir", '{trial.working_dir}']
+
+    orion.core.cli.main([
+        "hunt", "--config", "./orion_config.yaml", "./black_box.py"] + user_args)
+
+    exp = list(database.experiments.find({'name': 'voila_voici'}))
+    exp = exp[0]
+    exp_id = exp['_id']
+    trials = list(database.trials.find({'experiment': exp_id}))
+    trials = list(sorted(trials, key=lambda trial: trial['submit_time']))
+    params = trials[-1]['params']
+
+    assert (params[0]['value'] - 34.568) < 1e-5
```

Comment:
I would simply test that values of params have the proper precision.

Reply:
En fait `params[0]['value']` a une valeur de 34.56795111111093, ce qui est bien plus que demandé. Si je comprends bien la sélection des x est faite par gradient descent dans ce cas, donc j'imagine que ça ignore la précision demandé ?

Reply:
oh :scream: We just catched an issue. If the algo outputs in lower precision, it should be casted in transformations. But what we do right now is cast it in `transform`, which is before sending to algo, but we do nothing in `reverse`. We should cast in both directions.

Reply:
The modif to `reverse` causes another problem: `RuntimeError: Algorithm could not sample new points in less than 60 seconds`, because I guess it really wants to sample at values in between the precision